24_find_lab/findlab.txt

```diff
@@ -1,24 +1,140 @@
 1. Write command lines to find using "find":
 
   -. All files owned by yourself in /tmp folder
+  
+  all files  in /tmp/(2 users : maxim & test)
+  
+  maxim@maxim-virtual-machine:/tmp$ ls -l
+total 12
+-rw-rw-r-- 1 test  test     0 Jan  4 09:48 file1.test
+-rw-rw-r-- 1 test  test     0 Jan  4 09:48 file2.test
+drwx------ 3 root  root  4096 Jan  4 09:40 systemd-private-07794fef300641d3a64d3c789b0d3e70-colord.service-QhXkz8
+drwx------ 3 root  root  4096 Jan  4 09:40 systemd-private-07794fef300641d3a64d3c789b0d3e70-rtkit-daemon.service-cEkNto
+drwx------ 3 root  root  4096 Jan  4 09:39 systemd-private-07794fef300641d3a64d3c789b0d3e70-systemd-timesyncd.service-JiwDox
+-rw-rw-r-- 1 maxim maxim    0 Jan  4 09:42 test1.maxim
+-rw-rw-r-- 1 maxim maxim    0 Jan  4 09:42 test2.maxim
+-rw-rw-r-- 1 maxim maxim    0 Jan  4 09:42 test.maxim
+
+
+
+  maxim@maxim-virtual-machine:/tmp$ find . -exec ls -l \; | grep $(whoami) | uniq
+  
+  ** I don't understabd why it print all 4 times, and why "uniq" is not working.
```

Comment:
it will invoke ```ls -l``` each time a file is matched (and in this case for every file!)... 

as for ```uniq``` note that this command **by default** will remove identical and adjacent lines, here watch this:
```
vitali@Tux:~/tmp$ cat demo.txt 
wow!
wow!
wow!
moooo!
wow!
vitali@Tux:~/tmp$ uniq demo.txt 
wow!
moooo!
wow!
vitali@Tux:~/tmp$ 
```

you need to pass the ```-u``` flag, if you want unique lines:
```
vitali@Tux:~/tmp$ uniq -u demo.txt 
moooo!
wow!
```